src/core/PasswordGenerator.cpp

```diff
@@ -20,49 +20,41 @@
 
 #include "crypto/Random.h"
 
-const char* PasswordGenerator::DefaultAdditionalChars = "";
+const int PasswordGenerator::DefaultLength = 32;
+const char* PasswordGenerator::DefaultCustomCharacterSet = "";
 const char* PasswordGenerator::DefaultExcludedChars = "";
 
 PasswordGenerator::PasswordGenerator()
-    : m_length(0)
-    , m_classes(nullptr)
-    , m_flags(nullptr)
-    , m_additional(PasswordGenerator::DefaultAdditionalChars)
+    : m_length(PasswordGenerator::DefaultLength)
+    , m_classes(PasswordGenerator::CharClass::DefaultCharset)
+    , m_flags(PasswordGenerator::GeneratorFlag::DefaultFlags)
+    , m_custom(PasswordGenerator::DefaultCustomCharacterSet)
     , m_excluded(PasswordGenerator::DefaultExcludedChars)
 {
 }
 
 void PasswordGenerator::setLength(int length)
 {
-    if (length <= 0) {
-        m_length = DefaultLength;
-        return;
-    }
     m_length = length;
 }
 
-void PasswordGenerator::setCharClasses(const CharClasses& classes)
+void PasswordGenerator::setCharClasses(const PasswordGenerator::CharClasses& classes)
 {
-    if (classes == 0) {
-        m_classes = DefaultCharset;
-        return;
-    }
```

Comment:
I think you removed this because of the discussion surround custom and excluded characters. Perhaps it is better to add those as parameters to this function call instead of making them separate and requiring additional logic outside this class to ensure everything works properly.